# ==============================================================================
# REGRESSOR COMMIT DIFF
# ==============================================================================
# File: third_party/rust/md-5/tests/mod.rs
# Commit: 63b555e1f503
# Full Hash: 63b555e1f503db91a54604fd8cef88dfab7c0c7a
# Author: Dana Keeler <dkeeler@mozilla.com>
# Date: 2023-06-08 09:27:30
# Regressor Bug: 1834862
# File Overlap Count: 1
# Description:
#   Bug 1834862 - implement nsICryptoHash in rust r=jschanck,supply-chain-reviewers
#   
#   Differential Revision: https://phabricator.services.mozilla.com/D179727
# ==============================================================================

diff -r 6337ce701771 -r 63b555e1f503 third_party/rust/md-5/tests/mod.rs
--- /dev/null	Thu Jan 01 00:00:00 1970 +0000
+++ b/third_party/rust/md-5/tests/mod.rs	Wed Jun 07 23:03:21 2023 +0000
@@ -0,0 +1,15 @@
+use digest::dev::{feed_rand_16mib, fixed_reset_test};
+use hex_literal::hex;
+use md5::{Digest, Md5};
+
+digest::new_test!(md5_main, "md5", md5::Md5, fixed_reset_test);
+
+#[test]
+fn md5_rand() {
+    let mut h = Md5::new();
+    feed_rand_16mib(&mut h);
+    assert_eq!(
+        h.finalize()[..],
+        hex!("61aec26f1b909578ef638ae02dac0977")[..]
+    );
+}